spyder/app/mainwindow.py

```diff
@@ -604,6 +618,9 @@ def setup(self):
         ui_theme = CONF.get('appearance', 'ui_theme')
         color_scheme = CONF.get('appearance', 'selected')
 
+        logger.info('Deleting previous Spyder instance LSP logs...')
+        delete_lsp_log_files()

```

Comment:
Let's move this to the method `post_visible_setup` to not delay the window creation by this.